File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/module/ModuleRepository.java
package br.com.deltaglobalbank.identity.domain.module;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface ModuleRepository {
    Module findById(UUID id);

    Module findByCode(ModuleCode code);

    List<Module> findAll();

    List<Module> findAllByIds(Set<UUID> ids);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/module/ModuleRepository.kt
package br.com.deltaglobalbank.identity.domain.module

import java.util.UUID

interface ModuleRepository {
    fun findById(id: UUID): Module?
    fun findByCode(code: ModuleCode): Module?
    fun findAll(): List<Module>
    fun findAllByIds(ids: Set<UUID>): List<Module>
}
